peps/pep-0780.rst

```diff
@@ -0,0 +1,469 @@
+
+The Python standard library already has one mechanism to provide information
+about the ABI of the interpreter, namely the ``sys.abiflags`` attribute.
+This attribute is a string that contains a number of flags, such as ``d`` for
+debug builds, but it is not easily understood and not uniformly available.
+The new ``sys.abi_features`` attribute is both and is designed to be a superset
+of the information contained in ``sys.abiflags``, which is why we chose to
+include the ``debug`` feature.
+The established ``sys.abiflags`` also has a direct link to the naming of
+various filesystem paths. This can be convenient, but precludes its use for the
+documentation of important ABI features that are *not* intended to impact these
+paths.
+
+The new ``sys.abi_features`` attribute provides a place for this, and because
+it decouples the information from filesystem paths, it can offer significantly
+more clarity.
+
+A Forward Looking View on Free-Threaded Python
+----------------------------------------------
+
+:pep:`703`, which is the accepted proposal for free threading, states that the
+rollout of free-threaded Python should be gradual, which has been clarified
+by the Python Steering Council in `the PEP 703 acceptance post`_ to mean a
+three stage process over multiple releases. It is therefore important to make
+sure that the mechanisms in this PEP are useable for Python interpreters where
+either free-threading or non-free-threading could be the default or the only
+option.
+
+At the time of writing, free-threaded Python is in Phase I: experimental phase.
+In this phase, there is an acute need for the proposed environment markers to
+help with the transition to free-threaded Python as package authors gradually
+add support.
+
+As the number of packages with support increases, and particularly during
+Phase II: Supported-but-not-default phase, we still anticipate a strong need
+for the environment markers to help with the transition.
+
+As free-threaded Python enters into Phase III: Default phase, the need for
+the environment markers will decrease, though at this point it is not clear
+that the GIL-enabled Python will be completely phased out (it may remain
+available as a non standard build option). If it persists, the inverse need for
+the ABI feature detection may arise.
+
+Indeed, in all three phases it may be necessary for package authors to choose
+specific versions of their dependencies based on the ABI features, with a shift
+from GIL-enabled as default to free-threading as default over time.
+
+The ABI features are designed with this in mind to guarantee usefulness and
+simplicity for the foreseeable future in a changing Python ecosystem.
+
+Specification
+=============
+
+The keywords "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**",
+"**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**",
+and "**OPTIONAL**"" in this document are to be interpreted as described in
+:rfc:`RFC 2119 <2119>`.
+
+Teaser
+------
```

Comment:
I'm not sure a "Teaser" section should be in the Specification

Reply:
Agreed. With the more developed example section now, I think we can just remove this section here.